src/simsopt/mhd/spec.py

```diff
@@ -1029,10 +1028,10 @@ def run(self, update_guess: bool = True):
             si.curtor = si.ivolume[self.nvol - 1] + np.sum(si.isurf)
 
         if self.iota_profile is not None:
-            si.iota[0:self.nvol+1] = self.iota_profile.get(np.arange(0, self.nvol))
+            si.iota[0:self.nvol+1] = self.iota_profile.get(np.arange(0, self.nvol+1))
 
         if self.oita_profile is not None:
-            si.oita[0:self.nvol+1] = self.oita_profile.get(np.arange(0, self.nvol))
+            si.oita[0:self.nvol+1] = self.oita_profile.get(np.arange(0, self.nvol+1))
```

Comment:
nvol+1 is the correct size here, compare to:

https://github.com/missing-user/SPEC/blob/6a231606a626968fc1b83eed7dfa01c234842905/InputFiles/Verification/forcefree/solovev/generate_spec_nl_from_vmec.m#L167

all other iota sizes in spec wrapper must be adjusted